### PR TITLE
[proposal] Add update to revision

### PR DESCRIPTION
**Pull request by kesselborn**
We want to have an atomic rev update for bazooka cli when updating stacks

@grobie

### CODE REVIEW

File: revision.go [hunk 1]
```diff
@@ -71,6 +71,37 @@ func (r *Revision) Register() (*Revision, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	exists, _, err := sp.Exists(r.dir.Name)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("hihihihi")
```
Comment (xla) [inline revision.go]: Can we have a proper error here?

**Comment by xla** [comment]
The intention is understood, it's simply not atomic by virtue of lack of txns in dozoerd, besides this method makes sense. But we should talk about the update of the `Registered` timestamp, as it might be appropriate to have a different timestamp storing the latest update time.

**Comment by grobie** [comment]
As discussed internally already, I feel very unsure about being able to update revisions afterwards. We always treated them as immutable object for a good reason.

My proposal would be actually to extend the concept of the `*`-"HEAD" pointer to allow for any kind of alias name. An example list of the stack revisions would then look like:

```
TAG        REVISION   ARCHIVE-URL                         REGISTERED
base       b808767    http://url.com/bazooka/b808767.img  2015-06-30 15:32:12
jdk-8      5486804    http://url.com/bazooka/5486804.img  2015-06-30 15:41:44
ruby-1.9   f7d0936    http://url.com/bazooka/f7d0936.img  2015-06-30 15:44:27
           256afd4    http://url.com/bazooka/256afd4.img  2015-06-30 15:49:36
jdk-7      95aca01    http://url.com/bazooka/95aca01.img  2015-06-30 15:59:12
ruby-2.0   28925c2    http://url.com/bazooka/28925c2.img  2015-06-30 16:04:00
jruby      3937b97    http://url.com/bazooka/3937b97.img  2015-06-30 16:04:04
ruby-2.1   ecf8021    http://url.com/bazooka/ecf8021.img  2015-06-30 16:07:08
```

Moving the tag to a new revision would also be an atomic operation.

File: revision.go (hunk 1 above)
Comment (kesselborn) [inline revision.go]: hehe ... yeah, sure. sorry that this wasn't written in this PR: this was a proposal. If we agree on this functionality we were going to add a proper error + tests

**Comment by kesselborn** [comment]
@grobie could you guess how much work this would be and when this could be done? I agree that this is a much better approach just wasn't sure if there are enough resources to implement the tag approach.
Functional wise this PR does not add anything which is not already possible already: it just makes the `register` a revision which was previously `unregistered` atomic

**Comment by grobie** [comment]
I took a look at our various bazooka code bases earlier today and I feel more and more certain we should continue to treat revisions as immutable and remove any kind of revision naming which is not specific for a specific build (e.g. `HEAD`, `jdk-8`, etc. are all bad).

One additional example I found today is that the pms will never download an updated stack image otherwise, as it already has a `jdk-8` image on disk. So while it is true that is possible to forcefully break the guards around immutable revisions with `unregister && register`, this also doesn't work currently with unexpected and undefined side effects.

**Comment by grobie** [comment]
I also see a difference between "it does not add anything which is not already possible already" where "possible already" means actively working around the guards which are in place, and "bazooka actively encourages and provides functionality for a certain workflow".

**Comment by grobie** [comment]
@kesselborn I'll collect all necessary steps and will come back to you with an ETA as soon as possible.

**Comment by kesselborn** [comment]
@grobie as stated above: I think the tags are a much better solution, so we agree here.

**Comment by grobie** [comment]
:rainbow:

**Pull request closed without merging**